src/core/lib/iomgr/tcp_client_posix.cc

```diff
@@ -86,6 +100,41 @@ void do_tcp_client_global_init(void) {
   g_connection_shards = new std::vector<struct ConnectionShard>(num_shards);
 }
 
+int64_t event_engine_tcp_client_connect(
+    grpc_closure* on_connect, grpc_endpoint** endpoint,
+    const grpc_event_engine::experimental::EndpointConfig& config,
+    const grpc_resolved_address* addr, grpc_core::Timestamp deadline) {
+  auto resource_quota = reinterpret_cast<grpc_core::ResourceQuota*>(
+      config.GetVoidPointer(GRPC_ARG_RESOURCE_QUOTA));
+  auto addr_uri = grpc_sockaddr_to_uri(addr);
+  EventEngine::ConnectionHandle handle = GetDefaultEventEngine()->Connect(
+      [on_connect,
+       endpoint](absl::StatusOr<std::unique_ptr<EventEngine::Endpoint>> ep) {
+        grpc_core::ExecCtx exec_ctx;
+        absl::Status conn_status = ep.ok() ? absl::OkStatus() : ep.status();
+        if (ep.ok()) {
+          *endpoint = grpc_event_engine_endpoint_create(std::move(*ep));
+        } else {
+          *endpoint = nullptr;
+        }
+        grpc_core::ExecCtx::Run(DEBUG_LOCATION, on_connect,
+                                absl_status_to_grpc_error(conn_status));
```

Comment:
Is this conversion necessary? `typedef absl::Status grpc_error_handle;` is the latest state of things.

Reply:
Yes the absl_status_to_grpc_error function also sets the RPC status through an int property. A lot of tests fail without the RPC status.